extensions/amp-youtube/0.1/amp-youtube.js

```diff
@@ -71,6 +91,58 @@ class AmpYoutube extends AMP.BaseElement {
     // the playback.
     return false;
   }
+
+  /** @private */
+  buildImagePlaceholder_() {
+    const imgPlaceholder = new Image();
+    const videoid = this.videoid_;
+
+    setStyles(imgPlaceholder, {
+      // Cover matches YouTube Player styling.
+      'object-fit': 'cover',
+      // Hiding the placeholder initially to give the browser time to fix
+      // the object-fit: cover.
+      'visibility': 'hidden'
+    });
+
+    // Because sddefault.jpg isn't available for all videos, we try to load
+    // it and fallback to hqdefault.jpg.
+    loadPromise(imgPlaceholder).then(() => {
+      // Poll until complete is set.
+      if (!imgPlaceholder.complete) {
+        return whenDocumentReady(this.getWin().document);
+      }
+    }).then(() => {
+      // A pretty ugly hack since onerror won't fire on YouTube image 404.
```

Comment:
... ok ... after some more debugging ... I reached to the realization that the loadPromise is resolving pretty fast... That lead me to noticing that we are resolving the promise right away if `complete` is true ... and according to the `complete` specs, if the image doesn't have a `src` its `complete` is `TRUE`... 

Setting the `src` before calling `loadPromise` solves this... FML :sob: